src/main/cpp/src/genomicsdb/query_variants.cc

```diff
@@ -206,10 +210,22 @@ void VariantQueryProcessor::register_field_creators(const VariantArraySchema& sc
       m_field_factory.Register(i, KnownFieldInfo::get_field_creator(enumIdx));
     else
       m_field_factory.Register(i, (*iter).second);
+    //TileDB does not have a way to distinguish between char, string and int8_t fields
```

Comment:
In general, I think it's a good idea to distinguish between:
1. Fixed length char vector
1. Var length char vector (string)
1. Fixed/var length int8_t/uint8_t
even if all 3 types are stored internally as char types.
Distinguishing (1) and (2) probably may improve performance slightly. Keeping (3) separate is required for querying - for example, you could do something like regex matching for (1) and (2), but some integer arithmetic for (3). Just makes it easy for the system to check valid operations.
Largely borrowed from ideas in conventional DBs (SQLite, Postgres etc).